### PR TITLE
New sidebar component

### PR DESCRIPTION
**Pull request by balajimanoharan**
needs: #117 

New Sidebar component for the directory route
TODO 
- Plugin query params (future PR)

sidebar
sidebar1

### CODE REVIEW

File: packages/directory/tests/unit/controllers/directory-test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { moduleFor, test } from 'ember-qunit';
+
+moduleFor('controller:directory', 'Unit | Controller | directory');
```
Comment (Xuefeng-Zhu) [inline packages/directory/tests/unit/controllers/directory-test.js]: Do we want to keep this test?

File: packages/directory/tests/unit/routes/directory-test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { moduleFor, test } from 'ember-qunit';
+
+moduleFor('route:directory', 'Unit | Route | directory', {
```
Comment (Xuefeng-Zhu) [inline packages/directory/tests/unit/routes/directory-test.js]: Do we want to keep this test?

File: packages/directory/tests/unit/controllers/directory-test.js (hunk 1 above)
Comment (balajimanoharan) [reply packages/directory/tests/unit/controllers/directory-test.js]: Can you make these comments in #117. This isn't related to this PR.

Comment (balajimanoharan) [reply packages/directory/tests/unit/controllers/directory-test.js]: Also yes, we want to keep these. Once there is logic in place, we could just add the tests here.

File: packages/directory/app/styles/navi-directory/components/dir-sidebar.less [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    padding: 15px 0 15px 20px;
+
+    &--icon {
+      .icon;
+    }
+
+    &--selected {
+      background: @navi-white;
+      color: @navi-gray-700;
+      border-left: 4px solid @text-blue;
```
Comment (mattgraham1995) [inline packages/directory/app/styles/navi-directory/components/dir-sidebar.less]: 🔤

File: packages/directory/app/styles/navi-directory/components/dir-sidebar.less [hunk 4]
```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2018, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+
+ @text-blue: #3697f2;
```
Comment (mattgraham1995) [inline packages/directory/app/styles/navi-directory/components/dir-sidebar.less]: Maybe we should add this to the colors.less file in core. It seems like we're using it for more than just text

File: packages/directory/tests/integration/components/dir-sidebar-test.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+
+moduleForComponent('dir-sidebar', 'Integration | Component | dir sidebar', {
```
Comment (mattgraham1995) [inline packages/directory/tests/integration/components/dir-sidebar-test.js]: We should start moving to the new testing format as shown here
https://github.com/emberjs/ember-qunit/blob/master/docs/migration.md

File: packages/directory/app/styles/navi-directory/components/dir-sidebar.less (hunk 4 above)
Comment (balajimanoharan) [reply packages/directory/app/styles/navi-directory/components/dir-sidebar.less]: Yes, I meant to change it. Will do.

File: packages/directory/tests/integration/components/dir-sidebar-test.js (hunk 5 above)
Comment (balajimanoharan) [reply packages/directory/tests/integration/components/dir-sidebar-test.js]: We are still on 2.18, i think these changes come with 3.x.
I ran the code-mods, no change

File: packages/directory/app/styles/navi-directory/components/dir-sidebar.less (hunk 3 above)
Comment (balajimanoharan) [reply packages/directory/app/styles/navi-directory/components/dir-sidebar.less]: You don't like `bcb`?

**Comment by cythrawll** [comment]
should we throw liquid fire at this?

**Comment by balajimanoharan** [comment]
> should we throw liquid fire at this?

We want the sidebar to be always open right? the only option for using liquid fire would be to animate the filters for each group, I guess we could add that, but I would like to do that as a different PR, and get the basic functionality out first.

File: packages/directory/addon/components/dir-sidebar.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import { computed, get } from '@ember/object';
+
+export default Component.extend({
+  layout,
+
+  classNames: [ 'dir-sidebar' ],
+
+  directories: Directories,
+
+  selectedDirectory: computed('directories', function() {
```
Comment (Xuefeng-Zhu) [inline packages/directory/addon/components/dir-sidebar.js]: Can we use `computed.oneWay('directories.0')`?

File: packages/directory/addon/templates/components/dir-sidebar.hbs [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{{!-- Copyright 2018, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
+<section class='dir-sidebar__groups'>
+  {{#each directories as | directory |}}
+    <div
+      class='dir-sidebar__group {{if (eq selectedDirectory directory) 'dir-sidebar__group--selected'}}'
+      onclick={{pipe (action (mut selectedDirectory) directory) (action (mut selectedFilter) (object-at 0 directory.filters))}}
```
Comment (Xuefeng-Zhu) [inline packages/directory/addon/templates/components/dir-sidebar.hbs]: is mut selectedFilter required?

Comment (balajimanoharan) [reply packages/directory/addon/templates/components/dir-sidebar.hbs]: Yes, each directory has its set of filters. Its not necessary given that the selected filter would exist for the selected directory. So resetting it.

**Pull request merged**